backends/cxxrtl/cxxrtl_backend.cc

```diff
@@ -2326,7 +2326,8 @@ struct CxxrtlWorker {
 		f << escape_c_string(data);
 	}
 
-	void dump_metadata_map(const dict<RTLIL::IdString, RTLIL::Const> &metadata_map) {
+	template <typename SmellsLikeDict>
```

Comment:
-1 on this cutesy naming. The "project nyan" stuff (wasn't it NGY?) got old quite a while ago and given how annoying it is to work with Yosys internals the names just make it worse on top.